File: sol/solution/0001-0100/0006. Zigzag Conversion/Solution.php
class Solution {
    /**
     * @param String $s
     * @param Integer $numRows
     * @return String
     */
    function convert($s, $numRows) {
        if ($numRows == 1 || strlen($s) <= $numRows) {
            return $s;
        }
        $rows = array_fill(0, $numRows, "");
        $curRow = 0;
        $down = false;
        for ($i = 0; $i < strlen($s); $i++) {
            $rows[$curRow] .= $s[$i];
            if ($curRow == 0 || $curRow == $numRows - 1) {
                $down = !$down;
            }
            $curRow += $down ? 1 : -1;
        }
        return implode("", $rows);
    }
}

File: sol/solution/0001-0100/0006. Zigzag Conversion/Solution.rs
impl Solution {
    pub fn convert(s: String, num_rows: i32) -> String {
        if num_rows == 1 || s.len() <= num_rows as usize {
            return s;
        }
        let mut rows = vec![String::new(); num_rows as usize];
        let mut cur_row = 0;
        let mut down = false;
        for c in s.chars() {
            rows[cur_row].push(c);
            if cur_row == 0 || cur_row == (num_rows - 1) as usize {
                down = !down;
            }
            cur_row = if down { cur_row + 1 } else { cur_row - 1 };
        }
        rows.concat()
    }
}